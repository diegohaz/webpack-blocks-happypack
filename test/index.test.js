import { createConfig } from '@webpack-blocks/webpack'
import babel from '@webpack-blocks/babel'
import { css } from '@webpack-blocks/assets'
import extractText from '@webpack-blocks/extract-text'
import prettyFormat from 'pretty-format'
import happypack from '../src'

jest.mock('os', () => ({
  cpus: () => [1, 2, 3],
}))

jest.mock('../src/utils', () => ({
  ...require.requireActual('../src/utils'),
  createRuleHash: () => 'foo',
}))

const extractTextPath = require.resolve('extract-text-webpack-plugin/dist/loader')

expect.addSnapshotSerializer({
  test: value => value && value.loader && value.loader === extractTextPath,
  print: value => prettyFormat({
    ...value,
    loader: 'foo',
  }),
})

describe('happypack', () => {
  test('babel', () => {
    const config = createConfig([
      happypack([
        babel(),
      ]),
    ])
    expect(config).toMatchSnapshot()
  })

  test('cssModules', () => {
    const config = createConfig([
      happypack([
        css.modules(),
      ]),
    ])
    expect(config).toMatchSnapshot()
  })

  test('extractText', () => {
    const config = createConfig([
      css(),
      happypack([
        extractText(),
      ]),
    ])
    expect(config).toMatchSnapshot()
  })

  test('not allowed loaders', () => {
    const config = createConfig([
      happypack([
        babel(),
      ], {
        loaders: ['css-loader'],
      }),
    ])
    expect(config).toMatchSnapshot()
  })
})
