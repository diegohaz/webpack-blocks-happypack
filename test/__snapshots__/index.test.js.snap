// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happypack babel 1`] = `
Object {
  "module": Object {
    "loaders": Array [],
    "rules": Array [
      Object {
        "exclude": Array [
          /node_modules/,
        ],
        "test": /\\\\\\.\\(js\\|jsx\\)\\$/,
        "use": Array [
          "happypack/loader?id=js-jsx-foo",
        ],
      },
    ],
  },
  "plugins": Array [
    HappyPlugin {
      "config": Object {
        "cache": true,
        "cacheContext": Object {
          "env": "test",
          "refresh": 0,
        },
        "compilerId": "default",
        "enabled": true,
        "id": "js-jsx-foo",
        "loaders": Array [
          "babel-loader?{\\"cacheDirectory\\":true}",
        ],
        "tempDir": ".happypack",
        "threadPool": Object {
          "compile": [Function],
          "isRunning": [Function],
          "size": 3,
          "start": [Function],
          "stop": [Function],
        },
        "threads": 3,
        "verbose": true,
      },
      "id": "js-jsx-foo",
      "name": "HappyPack",
      "state": Object {
        "baseLoaderRequest": "",
        "debug": false,
        "foregroundThreadPool": null,
        "loaders": Array [],
        "started": false,
        "verbose": false,
      },
    },
  ],
}
`;

exports[`happypack cssModules 1`] = `
Object {
  "module": Object {
    "loaders": Array [],
    "rules": Array [
      Object {
        "test": /\\\\\\.css\\$/,
        "use": Array [
          "style-loader",
          "happypack/loader?id=css-foo",
        ],
      },
    ],
  },
  "plugins": Array [
    HappyPlugin {
      "config": Object {
        "cache": true,
        "cacheContext": Object {
          "env": "test",
          "refresh": 0,
        },
        "compilerId": "default",
        "enabled": true,
        "id": "css-foo",
        "loaders": Array [
          "css-loader?importLoaders=1&localIdentName=[name]--[local]--[hash:base64:5]&modules",
        ],
        "tempDir": ".happypack",
        "threadPool": Object {
          "compile": [Function],
          "isRunning": [Function],
          "size": 3,
          "start": [Function],
          "stop": [Function],
        },
        "threads": 3,
        "verbose": true,
      },
      "id": "css-foo",
      "name": "HappyPack",
      "state": Object {
        "baseLoaderRequest": "",
        "debug": false,
        "foregroundThreadPool": null,
        "loaders": Array [],
        "started": false,
        "verbose": false,
      },
    },
  ],
}
`;

exports[`happypack extractText 1`] = `
Object {
  "module": Object {
    "loaders": Array [
      Object {
        "loaders": Array [
          "style-loader",
          "css-loader?importLoaders=1&localIdentName=[name]--[local]--[hash:base64:5]&modules",
        ],
        "test": /\\\\\\.css\\$/,
      },
    ],
    "rules": Array [
      Object {
        "exclude": undefined,
        "test": /\\\\\\.css\\$/,
        "use": Array [
          Object {
  "loader": "foo",
  "options": Object {
    "id": 1,
    "omit": 1,
    "remove": true,
  },
},
          Object {
            "loader": "style-loader",
          },
          Object {
            "loader": "happypack/loader?id=css-foo",
          },
        ],
      },
    ],
  },
  "plugins": Array [
    ExtractTextPlugin {
      "filename": "css/[name].[contenthash:8].css",
      "id": 1,
      "options": Object {},
    },
    HappyPlugin {
      "config": Object {
        "cache": true,
        "cacheContext": Object {
          "env": "test",
          "refresh": 0,
        },
        "compilerId": "default",
        "enabled": true,
        "id": "css-foo",
        "loaders": Array [
          "css-loader?importLoaders=1&localIdentName=[name]--[local]--[hash:base64:5]&modules",
        ],
        "tempDir": ".happypack",
        "threadPool": Object {
          "compile": [Function],
          "isRunning": [Function],
          "size": 3,
          "start": [Function],
          "stop": [Function],
        },
        "threads": 3,
        "verbose": true,
      },
      "id": "css-foo",
      "name": "HappyPack",
      "state": Object {
        "baseLoaderRequest": "",
        "debug": false,
        "foregroundThreadPool": null,
        "loaders": Array [],
        "started": false,
        "verbose": false,
      },
    },
  ],
}
`;
